src/cljs/org/broadinstitute/firecloud_ui/utils.cljs

```
@@ -94,34 +99,34 @@
-                                     (.setRequestHeader us-xhr "Accept" "application/json" )
-                                     (.send us-xhr))
-                              (on-done {:xhr xhr
+                           (if (= status-code 401)
+                             (let [us-xhr (js/XMLHttpRequest.)]
+                               (set! (.-withCredentials us-xhr) true)
+                               (.addEventListener us-xhr "loadend"
+                                 (fn []
+                                   (let [us-status (.-status us-xhr)
+                                         parsed-us-response (parse-json-string (.-responseText us-xhr))
+                                         user-info (keywordize-keys parsed-us-response)]
+                                     (cond
+                                       (and (= us-status 200) (not (:ldap (:enabled (:userInfo user-info)))))
+                                       (on-done {:xhr (adjust-xhr xhr)
+                                                 :status-code 403
+                                                 :success? false
+                                                 :status-text "Access Disabled"
+                                                 :get-parsed-response get-parsed-response})
+                                       ;TODO: Fix this with a real log-out once the login bug is fixed and logout is implemented.
+                                       (= us-status 401) (set! (-> js/window .-location) "/")
+                                       :else (on-done {:xhr xhr
+                                                       :status-code status-code
+                                                       :success? false
+                                                       :status-text (.-statusText xhr)
+                                                       :get-parsed-response get-parsed-response})))))
```

In this case, `status-code` is always 401 and `get-parsed-response` is the default unauthorized text which is misleading. I think it makes more sense to return the info from the secondary user-status request.